iphone/cli/commands/_buildModule.js

```diff
@@ -460,156 +459,202 @@ iOSModuleBuilder.prototype.buildModule = function buildModule(next) {
 
 	const xcodeBuildArgumentsForTarget = function (target) {
 		let args = [
+			'archive',
 			'-configuration', 'Release',
 			'-sdk', target,
-			'-UseNewBuildSystem=NO',
+			'-archivePath', path.join(this.projectDir, 'build', target + '.xcarchive'),
+			'-UseNewBuildSystem=YES',
 			'ONLY_ACTIVE_ARCH=NO',
-			'clean', 'build'
+			'BUILD_LIBRARY_FOR_DISTRIBUTION=YES',
+			'SKIP_INSTALL=NO',
 		];
+		const scheme = this.isFramework ? this.moduleIdAsIdentifier : this.moduleName;
+		args.push('-scheme');
+		args.push(scheme);
 
-		if (this.isFramework) {
-			args.push('-scheme');
-			args.push(this.moduleIdAsIdentifier);
-			args.push('CONFIGURATION_BUILD_DIR=' + path.join(this.projectDir, 'build', 'Release-' + target));
-		}
-
-		// Exclude arm64 architecture from simulator build in XCode 12+ - TIMOB-28042
-		if (target === 'iphonesimulator' && parseFloat(this.xcodeEnv.version) >= 12.0) {
-			args.push('EXCLUDED_ARCHS=arm64');
+		if (target === 'macosx') {
+			args.push('SUPPORTS_MACCATALYST=YES');
 		}
 
 		return args;
 	}.bind(this);
 
-	// 1. Create a build for the simulator
-	xcodebuildHook(xcBuild, xcodeBuildArgumentsForTarget('iphonesimulator'), opts, 'xcode-sim', () => {
-		// 2. Create a build for the device
-		xcodebuildHook(xcBuild, xcodeBuildArgumentsForTarget('iphoneos'), opts, 'xcode-dist', next);
+	this.cli.env.getOSInfo(function (osInfo) {
+		const macOsVersion = osInfo.osver;
+
+		// 1. Create a build for the simulator
+		xcodebuildHook(xcBuild, xcodeBuildArgumentsForTarget('iphonesimulator'), opts, 'xcode-sim', () => {
+			// 2. Create a build for device
+			xcodebuildHook(xcBuild, xcodeBuildArgumentsForTarget('iphoneos'), opts, 'xcode-dist', () => {
+				const osVersionParts = macOsVersion.split('.').map(n => parseInt(n));
+				if (osVersionParts[0] > 10 || (osVersionParts[0] === 10 && osVersionParts[1] >= 15)) {
+					// 3. Create a build for the maccatalyst
+					xcodebuildHook(xcBuild, xcodeBuildArgumentsForTarget('macosx'), opts, 'xcode-macos', next);
```

Comment:
building module for macOS is intentional?

Reply:
Yes, I wanted to future proof. We do still plan on landing macOS if we can. But if we don't this would avoid having to rebuild the module multiple times.